modules/common/src/main/scala/notebook/front/widgets/package.scala

```diff
@@ -273,9 +275,25 @@ package object widgets {
     encoded
   } else Nil
 
+  def fromDFToPoint(df:DataFrame, maxPoints:Int):Seq[MagicRenderPoint] = {
+    val rows = df.take(maxPoints)
+    if (rows.nonEmpty) {
+      val points = df.schema.toList.head match {
```

Comment:
the case with rows is necessary in case the df is empty, which could happen.

`df.toJSON` is interesting to use, the sad thing is that it returns a `RDD[String]` so it''s lazy and we loose metadata (schema). It was hard to squeeze it literally and keeping the code simple (again, for a first cut)